### PR TITLE
[Checkpoint][Test] Add test for optimizer state_dict and resharding to 2d checkpoint test

### PR DESCRIPTION
**Pull request by wz337**
This PR updates the 2d checkpoint model state test to include:
1. optimizer state dict test
2. simple resharding test  (pg change)
3. rename test

### CODE REVIEW

File: test/distributed/checkpoint/test_2d_fsdp_dt_checkpoint.py [hunk 1]
```diff
@@ -171,6 +208,29 @@ def test_2d_model_state_checkpoint(self) -> None:
                     else:
                         self.assertEqual(n_p1[1], n_p2[1])
 
+        def opt_at(opt, idx):
+            return list(iter(opt.state.values()))[idx]
```
Comment (fduwjj) [inline test/distributed/checkpoint/test_2d_fsdp_dt_checkpoint.py:212]: Why do we need "iter" here?

**Review by fduwjj — APPROVED** [review]
LGTM. Unblock and have a small question here.

File: test/distributed/checkpoint/test_2d_fsdp_dt_checkpoint.py (hunk 1 above)
Comment (wz337) [reply test/distributed/checkpoint/test_2d_fsdp_dt_checkpoint.py:212]: Thanks for catching. iter is unnecessary here.

**Pull request closed without merging**